presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -891,6 +913,29 @@ public void dropTable(String databaseName, String tableName, boolean deleteData)
         }
     }
 
+    private static void deleteTableFilesIfNeeded(HdfsContext context, HdfsEnvironment hdfsEnvironment, String databaseName, String tableName, Path tablePath)
+    {
+        try {
+            if (isS3FileSystem(context, hdfsEnvironment, tablePath)) {
```

Comment:
@electrum @guyco33 : shouldn't this method keep up with the new changes for HADOOP_DEFAULT FS type in : https://github.com/prestosql/presto/commit/1ab3978b9744534c10c116c5337d8f298bd04b68 ?

Reply:
@electrum It looks like it's going to be skipped
https://github.com/prestosql/presto/blob/57d396b2631dac6bb622a3632d5c935460a64db9/presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriteUtils.java#L470-L479

Reply:
Agreed, although this an existing problem separate from this PR. cc @apc999

Reply:
Can we rely on `path` parameter for detecting S3 file system?
For example:
- `path.getScheme().equals("s3")`
- `fileSystem.getUri().getScheme().equals("s3")`